### PR TITLE
WIP: EasyMDE

### PR DESCRIPTION
**Pull request by jolheiser**
Convert SimpleMDE to EasyMDE

As far as I can tell this works (haven't done a full test yet), but perhaps it's not ideal.

Ideally this would be done with npm, however I don't understand nearly enough of that wizardry to get it working. We need not only `easymde` and `codemirror`, but _all_ of codemirror's addons/modes.

Since SimpleMDE wasn't _too_ pervasive yet, I decided to also rename variables...

### CODE REVIEW

**Comment by silverwind** [comment]
Thanks for doing this.

> Ideally this would be done with npm, however I don't understand nearly enough of that black wizardry to get it working

Vendoring is fine. We don't have a proper concept to lazy-load based on server's decision yet, but I think I will figure out something and then it can be easily un-vendored.

File: templates/base/footer.tmpl [hunk 1]
```diff
@@ -14,8 +14,9 @@
 
 	<script src="{{StaticUrlPrefix}}/js/jquery.js?v={{MD5 AppVer}}"></script>
 	<script src="{{StaticUrlPrefix}}/vendor/plugins/jquery.areyousure/jquery.are-you-sure.js"></script>
-{{if .RequireSimpleMDE}}
-	<script src="{{StaticUrlPrefix}}/vendor/plugins/simplemde/simplemde.min.js"></script>
+{{if .RequireEasyMDE}}
+	<script src="{{StaticUrlPrefix}}/vendor/plugins/easymde/easymde.min.js"></script>
+	<script src="{{StaticUrlPrefix}}/vendor/plugins/codemirror/codemirror.js"></script>
```
Comment (silverwind) [inline templates/base/footer.tmpl:19]: Interesting. Why did it work without CodeMirror being loaded before?

Comment (jolheiser) [reply templates/base/footer.tmpl:19]: To be perfectly honest I have no clue, but there is an open PR (that looks unlikely to be merged) regarding it.  
https://github.com/Ionaru/easy-markdown-editor/pull/76

Comment (silverwind) [reply templates/base/footer.tmpl:19]: I see. I think their method of loading CodeMirror would work in webpack but with this method, we need to manually load it.

Comment (silverwind) [reply templates/base/footer.tmpl:19]: I guess you should swap the order of these script tags thought.

**Comment by silverwind** [comment]
Needs a rebase.

**Comment by zeripath** [comment]
@jolheiser I hope you don't mind I rebased for you.

**Comment by zeripath** [comment]
I can't seem to be able to get a new line on the current head.

**Comment by silverwind** [comment]
FYI regarding webpack: I attempted to webpack-build simplemde, but I think at least the codemirror addons/modes are not bundleable because `loadmode` does `require(file)` in the middle of it's code which webpack can not process.

What may work is to just copy the CM files and set `modeURL` and write a custom version of `loadmode` in our code that does `jQuery.getScript()` or similar.

**Comment by silverwind** [comment]
Related: https://github.com/go-gitea/gitea/issues/10729

**Comment by silverwind** [comment]
SimpleMDE will be ripped out eventually, so I guess we can close this.

**Comment by jolheiser** [comment]
Agreed, closing this.

**Pull request closed without merging**